fallback_builtins.h

```diff
@@ -46,11 +46,13 @@ static __forceinline unsigned long long __builtin_ctzll(uint64_t value) {
 #endif // Microsoft AMD64/IA64/x86/ARM/ARM64 test
 #endif // _MSC_VER & !clang
 
-/* Unfortunately GCC didn't support these things until version 10 */
+/* Unfortunately GCC didn't support these things until version 10.
+ * Similarly, AppleClang didn't support them in Xcode 9.2 but did in 9.3.
+ */
 #ifdef __AVX2__
 #include <immintrin.h>
 
-#if (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 10)
+#if (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 10) || (defined(__apple_build_version__) && __apple_build_version__ < 9020039)
```

Comment:
I am wondering if after the `||` there should be a new line break and the next line align the `(`.

Reply:
Personally I like to move the `||` and have that start the next line, since I think that makes it more readable.
Like this:
```
#if (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 10)
    || (defined(__apple_build_version__) && __apple_build_version__ < 9020039)
```

Reply:
Why are you checking the XCode version instead of the AppleClang version?

Reply:
The line is clearly longer than 120 characters, so should be split.

@phprus Because checking just AppleClang version would match stock Clang too...

Reply:
I've never participated in this project, and I don't know your coding standards, and I don't know which preference here is authoritative.  You don't have some kind of clang-tidy-auto-format-on-merge script?  I can modify the commit formatting, but I'd rather have to do it only once. :)  Which way must it be?

Reply:
If everyone says different things, it's safe bet to do what @Dead2 says as he's the maintainer...